src/ConnectedMode/SlCoreConnectionAdapter.cs

```diff
@@ -0,0 +1,106 @@
+public class SlCoreConnectionAdapter : ISlCoreConnectionAdapter
+{
+    private readonly ISLCoreServiceProvider serviceProvider;
+    private readonly IThreadHandling threadHandling;
+    private readonly ILogger logger;
+
+    [ImportingConstructor]
+    public SlCoreConnectionAdapter(ISLCoreServiceProvider serviceProvider, IThreadHandling threadHandling, ILogger logger)
+    {
+        this.serviceProvider = serviceProvider;
+        this.threadHandling = threadHandling;
+        this.logger = logger;
+    }
+
+    public async Task<ValidateConnectionResponse> ValidateConnectionAsync(ConnectionInfo connectionInfo, string token)
+    {
+        var validateConnectionParams = GetValidateConnectionParams(connectionInfo, GetEitherForToken(token));
+        return await ValidateConnectionAsync(validateConnectionParams);
+    }
+
+    public async Task<ValidateConnectionResponse> ValidateConnectionAsync(ConnectionInfo connectionInfo, string username, string password)
+    {
+        var validateConnectionParams = GetValidateConnectionParams(connectionInfo, GetEitherForUsernamePassword(username, password));
+        return await ValidateConnectionAsync(validateConnectionParams);
+    }
+
+    private async Task<ValidateConnectionResponse> ValidateConnectionAsync(ValidateConnectionParams validateConnectionParams)
+    {
+        await threadHandling.SwitchToBackgroundThread();
+
+        if (!serviceProvider.TryGetTransientService(out IConnectionConfigurationSLCoreService connectionConfigurationSlCoreService))
+        {
+            logger.LogVerbose($"[{nameof(IConnectionConfigurationSLCoreService)}] {SLCoreStrings.ServiceProviderNotInitialized}");
+            return new ValidateConnectionResponse(false, Resources.ValidateCredentials_Fails);
+        }
+
+        try
+        {
+            return await connectionConfigurationSlCoreService.ValidateConnectionAsync(validateConnectionParams);
+        }
+        catch (Exception)
+        {
+            return new ValidateConnectionResponse(false, Resources.ValidateCredentials_Fails);
+        }
+    }
+
+    private static ValidateConnectionParams GetValidateConnectionParams(ConnectionInfo connectionInfo, Either<TokenDto, UsernamePasswordDto> credentials)
+    {
+        return new ValidateConnectionParams(GetTransientConnectionDto(connectionInfo, credentials));
+    }
+
+    private static Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> GetTransientConnectionDto(ConnectionInfo connectionInfo, Either<TokenDto, UsernamePasswordDto> credentials)
+    {
+        return connectionInfo.ServerType == ConnectionServerType.SonarQube
+            ? Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto>.CreateLeft(new TransientSonarQubeConnectionDto(connectionInfo.Id, credentials))
+            : Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto>.CreateRight(new TransientSonarCloudConnectionDto(connectionInfo.Id, credentials));
```

Comment:
if there's no organization, we should provide `null` here. Is that the case? Or do we provide `https://sonarcloud.io` here then?

Reply:
Yes this is actually an issue, it must be `null`

Reply:
Here we provide https://sonarcloud.io. I have to change it to null, but in another part of the code.

Reply:
I will do this on the UI PR, because it is also possible to test. Additionally, I would have to adapt the ConnectedInfoComponent to show the sonarcloud.io when the organization key is NULL